src/index.js

```diff
@@ -436,120 +412,70 @@ export var Lazy = function(props) {
 }
 
 export var h = function(name, props) {
-  props = props || EMPTY_OBJECT
-  var node
-  var rest = []
-  var children = []
-  var length = arguments.length
-
-  while (length-- > 2) rest.push(arguments[length])
+  for (var node, rest = [], children = [], i = arguments.length; i-- > 2; ) {
+    rest.push(arguments[i])
+  }
 
   while (rest.length > 0) {
     if (isArray((node = rest.pop()))) {
-      for (length = node.length; length-- > 0; ) {
-        rest.push(node[length])
-      }
+      for (i = node.length; i-- > 0; ) rest.push(node[i])
     } else if (node === false || node === true || node == null) {
     } else {
       children.push(typeof node === "object" ? node : createTextVNode(node))
     }
   }
 
+  props = props || EMPTY_OBJECT
+
   return typeof name === "function"
     ? name(props, children)
     : createVNode(name, props, children, null, props.key, DEFAULT_NODE)
 }
 
-var isSameAction = function(a, b) {
-  return isArray(a) && isArray(b) && typeof a[0] === "function" && a[0] === b[0]
-}
-
-var shouldRestart = function(a, b) {
-  for (var k in merge(a, b)) {
-    if (a[k] === b[k] || isSameAction(a[k], b[k])) b[k] = a[k]
-    else return true
-  }
-}
-
-var patchSub = function(oldSub, newSub, dispatch) {
-  if (
-    (newSub && (!newSub[0] || isArray(newSub[0]))) ||
-    (oldSub && (!oldSub[0] || isArray(oldSub[0])))
-  ) {
-    var subs = []
-    var newSubs = newSub ? newSub : [newSub]
-    var oldSubs = oldSub ? oldSub : [oldSub]
-
-    for (var i = 0; i < newSubs.length || i < oldSubs.length; i++) {
-      subs.push(patchSub(oldSubs[i], newSubs[i], dispatch))
-    }
-
-    return subs
-  }
-
-  return newSub
-    ? !oldSub || newSub[0] !== oldSub[0] || shouldRestart(newSub[1], oldSub[1])
-      ? [
-          newSub[0],
-          newSub[1],
-          newSub[0](newSub[1], dispatch),
-          oldSub && oldSub[2]()
-        ]
-      : oldSub
-    : oldSub && oldSub[2]()
-}
-
-export function app(props) {
+export var app = function(props) {
   var container = props.container
   var element = container && container.children[0]
-  var oldNode = element && recycleElement(element)
+  var node = element && recycleElement(element)
   var subs = props.subscriptions
   var view = props.view
-  var renderLock = false
-  var state
-  var sub
+  var lock = false
+  var state = {}
+  var sub = []
+
+  var eventCb = function(event) {
+    dispatch(event.currentTarget.events[event.type], event)
+  }
 
   var setState = function(newState) {
-    if (!(state === newState || renderLock)) {
-      defer(render, (renderLock = true))
+    if (!(state === newState || lock)) {
+      defer(render, (lock = true))
     }
     state = newState
   }
 
   var dispatch = function(obj, props) {
-    if (obj == null) {
```

Comment:
Was this removed purely for size?

I was use/abusing that null check to avoid needing to wrap `dispatch` calls in conditionals sometimes.

Reply:
@SkaterDad 

> Was this removed purely for size?

Partly, yes. The other reason is that I don't think there is a use case for returning `null` or `undefined` from an action. Is there?

> I was use/abusing that null check to avoid needing to wrap dispatch calls in conditionals sometimes.

Show me an example, please. 🙏

Reply:
Before the recent commits, I could do this:
```js
function myEffect(props, dispatch) {
  // props.init could be undefined.
  // but dispatch was checking for `obj == null`, so it was okay
  dispatch(props.init);

 // do other stuff
  setTimeout(() => {
    dispatch(props.action);
  }, 100)
}
```

Now my effect code will have to to wrap optional props in an `if...then` statement:
```js
function myEffect(props, dispatch) {
  // props.init could be undefined.
  // If we don't check for truthiness here, dispatch will end up doing `setState(undefined)`
  if (props.init) {
    dispatch(props.init);
  }

 // do other stuff
  setTimeout(() => {
    dispatch(props.action);
  }, 100)
}
```

Reply:
@SkaterDad Ah, I see. Thanks. Convenient, but maybe not so if you needed to do something when `props.init` is `undefined` instead. In that case, we'll end up with two checks: Hyperapp's and yours. I'll note this on the documentation at some point. 👍